pyiron/sphinx/base.py

```diff
@@ -1515,6 +1522,93 @@ def write_spins_constraints(self, file_name="spins.in", cwd=None, spins_str=None
             s.logger.debug("No magnetic moments")
 
 
+class Group(dict):
+    def __init__(self, *args, **kw):
+        super(Group, self).__init__(*args, **kw)
+
+    def __setitem__(self, key, value):
+        if isinstance(value, dict):
+            value = Group(value)
+        super(Group, self).__setitem__(key, value)
+
+    def __setattr__(self, key, value):
+        if isinstance(value, dict):
+            value = Group(value)
+        super(Group, self).__setitem__(key, value)
+
+    __getattr__ = dict.get
+    # __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def set(self, name, content):
+        self[name] = content
+
+    def set_group(self, name, content=None):
+        if content is None:
+            content = Group()
+        self.set(name, content)
+
+    def set_flag(self, flag, val=True):
+        self.set(flag, val)
+
+    def set_parameter(self, parameter, val):
+        self.set(parameter, val)
```

Comment:
wouldn't a simple
```set(whatever, content)```
not be simpler and then use the content to do the right thing?
```
set ('structure', Group ()) # set a group
set ('dipoleCorrection', True) # set a flag
set ('nEmptyStates', 23) # set a parameter
```
Largely self-explaining, no need to think about the right function to use.
One could still argue whether flags should have the ```True``` content, or if `None` content corresponds to a flag set to true...

Reply:
`set ('structure', Group ())` won't be possible since `Group()` will be defined only inside this class and therefore not available on the notebook. But then I guess also it's a matter of taste whether `set_group('structure')` is better or `set_group('structure', Group())`, which for me is also an argument for `set('dipoleCorrection', True)` or `set_flag('dipoleCorrection')`.

Reply:
what about
```
set ('structure{}')
set ('dipoleCorrection')
set ('nEmptyStates', 23)
````
? 
But I don't insist on this.

Reply:
I like the way @ashtonmv did it, but I don't insist either.

Reply:
> set ('structure', Group ()) # set a group
> set ('dipoleCorrection', True) # set a flag
> set ('nEmptyStates', 23) # set a parameter

These are all already possible with the `set` function. (Except, as Sam pointed out, Group won't usually be imported into a user's notebook, but the same goal is achieved by `set('structure', {})`, since all dictionaries are internally converted to Groups within `__setitem__()`)

I personally like having the specifically-named methods along with `set`, so that if I'm just stabbing in the dark with autocomplete it's more clear what those methods would accomplish. `set` is programmatically pretty ambiguous, and if I saw it as an autocomplete option I would probably naively expect it to return a "set" of the Group's unique items. It's also nice to have defaults for `set_group` & `set_flag`, so I can just write
```python
job.input.structure.set_group("symmetry")
job.input.hamilton.set_flag("dipoleCorrection")
```
But I'm willing to remove them for purity's sake if I'm outnumbered!

Reply:
ok, I give in. Let's drop my suggestions for the setter routines (in particular since I realized only later that Sam and you had a previous discussion on this, and probably
```
set ('structure',{})
```
will also work, no?

Reply:
yep, that works